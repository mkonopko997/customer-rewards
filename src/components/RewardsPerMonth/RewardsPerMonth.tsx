import { FC } from "react";
import { Link, useParams } from "react-router-dom";
import { useRewardsPerMonth } from "@/hooks/useRewardsPerMonth/useRewardsPerMonth";
import {
  CircularProgress,
  Paper,
  Table,
  TableBody,
  TableCell,
  TableContainer,
  TableHead,
  TableRow,
} from "@mui/material";

export const RewardsPerMonth: FC = () => {
  const { customer } = useParams<{ customer: string }>();
  const rewardPerMonth = useRewardsPerMonth(customer);

  const getTable = () => {
    if (!rewardPerMonth) {
      return (
        <div className="w-full flex justify-center">
          <CircularProgress />
        </div>
      );
    }
    return (
      <TableContainer component={Paper}>
        <Table aria-label="simple table">
          <TableHead>
            <TableRow>
              <TableCell>Month</TableCell>
              <TableCell>Reward</TableCell>
            </TableRow>
          </TableHead>
          <TableBody>
            {Object.keys(rewardPerMonth).map((key) => (
              <TableRow key={key}>
                <TableCell>{key}</TableCell>
                <TableCell>{rewardPerMonth[key]}</TableCell>
              </TableRow>
            ))}
          </TableBody>
        </Table>
      </TableContainer>
    );
  };

  return (
    <div>
      <div className="flex justify-between mb-3">
        <Link to="../" className="text-blue-500 hover:underline">
          Back
        </Link>
        <div>Showing rewards per month for {customer}</div>
      </div>
      {getTable()}
    </div>
  );
};
