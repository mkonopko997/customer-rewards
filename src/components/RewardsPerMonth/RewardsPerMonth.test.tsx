import { render, waitFor } from "@/test/testUtils/testUtils";
import { screen } from "@testing-library/react";
import { RewardsPerMonth } from "@/components/RewardsPerMonth/RewardsPerMonth";

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useParams: jest.fn().mockReturnValue({ customer: "testCustomer1" }),
}));

describe("RewardsPerMonth", () => {
  it("should display table of months and rewards", async () => {
    const { container } = render(<RewardsPerMonth />);

    await waitFor(() => {
      expect(screen.queryByRole("progressbar")).not.toBeInTheDocument();
    });

    expect(container).toMatchSnapshot();
  });

  it("should display back link with correct href", async () => {
    render(<RewardsPerMonth />);

    await waitFor(() => {
      expect(screen.queryByRole("progressbar")).not.toBeInTheDocument();
    });

    expect(screen.getByRole("link", { name: "Back" })).toHaveAttribute(
      "href",
      "/"
    );
  });

  it("should display current customer label", async () => {
    render(<RewardsPerMonth />);

    await waitFor(() => {
      expect(screen.queryByRole("progressbar")).not.toBeInTheDocument();
    });

    expect(
      screen.getByText("Showing rewards per month for testCustomer1")
    ).toBeInTheDocument();
  });
});
