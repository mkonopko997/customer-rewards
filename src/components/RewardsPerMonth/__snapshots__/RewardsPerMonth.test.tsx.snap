// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RewardsPerMonth should display table of months and rewards 1`] = `
<div>
  <div>
    <div
      class="flex justify-between mb-3"
    >
      <a
        class="text-blue-500 hover:underline"
        href="/"
      >
        Back
      </a>
      <div>
        Showing rewards per month for 
        testCustomer1
      </div>
    </div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiTableContainer-root css-11xur9t-MuiPaper-root-MuiTableContainer-root"
    >
      <table
        aria-label="simple table"
        class="MuiTable-root css-rqglhn-MuiTable-root"
      >
        <thead
          class="MuiTableHead-root css-15wwp11-MuiTableHead-root"
        >
          <tr
            class="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
          >
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
              scope="col"
            >
              Month
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
              scope="col"
            >
              Reward
            </th>
          </tr>
        </thead>
        <tbody
          class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
        >
          <tr
            class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
          >
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
            >
              3-2023
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
            >
              0
            </td>
          </tr>
          <tr
            class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
          >
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
            >
              2-2023
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
            >
              240
            </td>
          </tr>
          <tr
            class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
          >
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
            >
              1-2023
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
            >
              160
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;
